msgraphcore/graph_session.py

```diff
@@ -1,21 +1,28 @@
 """
-Creates a session object
+Graph Session
 """
 from requests import Session, Request, Response
 
 from msgraphcore.constants import BASE_URL, SDK_VERSION
 from msgraphcore.middleware._middleware import MiddlewarePipeline, BaseMiddleware
+from msgraphcore.middleware.options.auth_middleware_options import AuthMiddlewareOptions
+from msgraphcore.middleware._base_auth import AuthProviderBase
+from msgraphcore.middleware.authorization_handler import AuthorizationHandler
 
 
 class GraphSession(Session):
     """
     Extends session object with graph functionality
     """
-    def __init__(self, **kwargs):
+    def __init__(self, scopes: [str], auth_provider: AuthProviderBase, middleware: list = []):
         super().__init__()
         self.headers.update({'sdkVersion': 'graph-python-' + SDK_VERSION})
         self._base_url = BASE_URL
-        middleware = kwargs.get('middleware')
+
+        options = AuthMiddlewareOptions(scopes)
+        auth_handler = AuthorizationHandler(auth_provider, auth_provider_options=options)
```

Comment:
auth_provider_options should be set on the auth_provider, not in the AuthorizationHandler. The AuthorizationHandler can access it from the auth_provider. I can't think of a reason for them to be decoupled.